plugins/command-manager/src/main/java/com/wazuh/commandmanager/rest/RestPostCommandAction.java

```diff
@@ -88,77 +86,59 @@ protected RestChannelConsumer prepareRequest(final RestRequest request, final No
      * Handles a POST HTTP request.
      *
      * @param request POST HTTP request
+     * @param client NodeClient instance
      * @return a response to the request as BytesRestResponse.
      * @throws IOException thrown by the XContentParser methods.
      */
-    private RestChannelConsumer handlePost(RestRequest request) throws IOException {
+    private RestChannelConsumer handlePost(RestRequest request, final NodeClient client)
+            throws IOException {
         log.info(
                 "Received {} {} request id [{}] from host [{}]",
                 request.method().name(),
                 request.uri(),
                 request.getRequestId(),
                 request.header("Host"));
-
-        /// Request validation
-        /// ==================
-        /// Fail fast.
+        // Request validation
         if (!request.hasContent()) {
-            // Bad request if body doesn't exist
             return channel -> {
                 channel.sendResponse(
                         new BytesRestResponse(RestStatus.BAD_REQUEST, "Body content is required"));
             };
         }
-
-        /// Request parsing
-        /// ===============
-        /// Retrieves and generates an array list of commands.
-        XContentParser parser = request.contentParser();
-        List<Command> commands = new ArrayList<>();
-        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-        // The array of commands is inside the "commands" JSON object.
-        // This line moves the parser pointer to this object.
-        parser.nextToken();
-        if (parser.nextToken() == XContentParser.Token.START_ARRAY) {
-            commands = Command.parseToArray(parser);
-        } else {
-            log.error("Token does not match {}", parser.currentToken());
+        List<Command> commands = getCommandList(request);
+        // Validate commands are not empty
+        if (commands.isEmpty()) {
+            return channel -> {
+                channel.sendResponse(
+                        new BytesRestResponse(
+                                RestStatus.BAD_REQUEST,
+                                "No valid commands detected in the request body."));
```

Comment:
"No commands provided" is shorter and translates better to the condition being checked.

Reply:
Fixed on de10364e17b61e068b2c0ac7be853ff2f8328436